src/hnsw.cpp

```diff
@@ -116,8 +116,8 @@ class Hnsw {
     // we can't touch R API in a threaded code, but we can use data
     // allocated by R
     // so we need to get a pointer to the underlying matrix data
-    uint32_t ncol = items.ncol();
-    uint32_t nrow = items.nrow();
+    int ncol = items.ncol();
```

Comment:
could you rather try to add `#include <stdint.h>` and see if it helps?

Reply:
It does not seem to help removing the warnings
```
   "C:/rtools40/mingw64/bin/"g++  -std=gnu++11 -I"C:/opt/R/R-40~1.0/include" -DNDEBUG -I../inst/include/ -I'C:/opt/R/R-4.0.0/library/Rcpp/include'     -fopenmp -DNO_MANUAL_VECTORIZATION -DSTRICT_R_HEADERS   -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign -UNDEBUG -Wall -pedantic -g -O0 -fdiagnostics-color=always -c hnsw.cpp -o hnsw.o
   hnsw.cpp: In instantiation of 'void Hnsw<dist_t, Distance, DoNormalize>::addItems(Rcpp::NumericMatrix, int) [with dist_t = float; Distance = hnswlib::L2Space; bool DoNormalize = false; Rcpp::NumericMatrix = Rcpp::Matrix<14>]':
   hnsw.cpp:329:34:   required from here
   hnsw.cpp:132:15:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
          #pragma omp for schedule(dynamic, 64)
                  ^~~
   hnsw.cpp:136:28:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
            for (auto j = 0; j < ncol; j ++) {
                             ~~^~~~~~
   hnsw.cpp: In instantiation of 'void Hnsw<dist_t, Distance, DoNormalize>::addItems(Rcpp::NumericMatrix, int) [with dist_t = float; Distance = hnswlib::InnerProductSpace; bool DoNormalize = true; Rcpp::NumericMatrix = Rcpp::Matrix<14>]':
   hnsw.cpp:349:38:   required from here
   hnsw.cpp:132:15:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
          #pragma omp for schedule(dynamic, 64)
                  ^~~
   hnsw.cpp:136:28:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
            for (auto j = 0; j < ncol; j ++) {
                             ~~^~~~~~
   hnsw.cpp: In instantiation of 'void Hnsw<dist_t, Distance, DoNormalize>::addItems(Rcpp::NumericMatrix, int) [with dist_t = float; Distance = hnswlib::InnerProductSpace; bool DoNormalize = false; Rcpp::NumericMatrix = Rcpp::Matrix<14>]':
   hnsw.cpp:369:36:   required from here
   hnsw.cpp:132:15:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
          #pragma omp for schedule(dynamic, 64)
                  ^~~
   hnsw.cpp:136:28:warning: mparison of integer expressions of different signedness: 'intand 'uint32_t{aka 'unsigned int-Wsign-compare]
            for (auto j = 0; j < ncol; j ++) {
                             ~~^~~~~~
   C:/rtools40/mingw64/bin/g++ -shared -s -static-libgcc -o RcppHNSW.dll tmp.def RcppExports.o hnsw.o -fopenmp -LC:/opt/R/R-40~1.0/bin/x64 -lRblas -LC:/opt/R/R-40~1.0/bin/x64 -lR
   installing to C:/Users/samgg/AppData/Local/Temp/Rtmp2X6kjo/devtools_install_137c171577c3/00LOCK-rcpphnsw/00new/RcppHNSW/libs/x64
-  DONE (RcppHNSW)
```

Reply:
Sorry, just want to clarify. Do you mean that you have `uint32_t nrow, ncol` and still you have warning "comparison of integer expressions of different signedness"  in the for loops?

Reply:
No problem to clarify. When I set uint32_t for ncol and nrow, I got the warnings above. When I set int, there are no warnings at all.

Reply:
strange! =)